akka-distributed-data/src/main/scala/akka/cluster/ddata/PruningState.scala

```diff
@@ -44,5 +46,5 @@ import akka.annotation.InternalApi
           this
     }
 
-  def addSeen(node: Address): PruningState = this
+  def addSeen(node: Address): PruningState
```

Comment:
Why couldn't it be concrete here?

Reply:
There are only 2 implementations of `PruningState`. One of them overrides this method, the other one used the default implementation. It seemed easier to follow (more symmetrical) to have the method abstract here and have 2 concrete implementations in the 2 subclasses.

The reason I looked at this at all is since 'node' is unused here. I didn't want to add an `@unused` annotation here, since that is misleading: the 'node' parameter *is* used in the 'other' implementation of the method. I sidestepped that problem by making the improvement above instead - not a general solution but an opportunistic one here.